src/parsers/row_parser.rs

```diff
@@ -60,34 +64,40 @@ pub fn parse(line: &str, state: &mut State, strategies: &Strategies) -> String {
             },
         ) => {
             state.update_position(Position::InCreateTable {
-                table_name,
+                table_name: table_name.clone(),
                 types: add_create_table_row_to_types(sanitised_line, current_types.to_vec()),
             });
-            line.to_string()
+            Cow::from(line)
         }
         (RowType::CreateTableEnd, _position) => {
             state.update_position(Position::Normal);
-            line.to_string()
+            Cow::from(line)
         }
         (RowType::CopyBlockStart, _position) => {
             let current_table = copy_row::parse(sanitised_line, strategies);
             state.update_position(Position::InCopy { current_table });
-            line.to_string()
+            Cow::from(line)
         }
         (RowType::CopyBlockEnd, _position) => {
             state.update_position(Position::Normal);
-            line.to_string()
+            Cow::from(line)
         }
-        (RowType::CopyBlockRow, Position::InCopy { current_table }) => {
+        (RowType::CopyBlockRow, Position::InCopy { ref current_table }) => {
```

Comment:
Switching the order of the `Cow::from(...)` and `state.update_position(...)` statements somehow appeases the borrow checker when we match on `&state.position` instead of `state.position.clone()` 🤷